app/components/sidebar-menu.tsx

```diff
@@ -125,18 +162,28 @@ export default function SidebarMenu({ onClose }: SidebarMenuProps) {
                         <Switch
                             className="ml-auto h-6 w-10 cursor-pointer p-1"
                             checked={theme === 'dark'}
-                            onCheckedChange={(checked) =>
-                                setTheme(checked ? 'dark' : 'light')
-                            }
+                            onCheckedChange={(checked) => {
+                                const newTheme = checked ? 'dark' : 'light';
+                                setTheme(newTheme);
+                                debouncedLogTheme(newTheme);
+                            }}
```

Comment:
See the comment/suggestion above
```suggestion
                            onCheckedChange={handleThemeSwitch}
```